camel-route-support/src/main/java/io/kaoto/backend/api/service/step/parser/camelroute/CamelRouteDeserializer.java

```diff
@@ -29,8 +30,12 @@ protected CamelRouteDeserializer(Class<?> vc) {
     @Override
     public CamelRoute deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
         CamelRoute answer = new CamelRoute();
-        ArrayNode root = (ArrayNode) p.getCodec().readTree(p);
-        Iterator<JsonNode> fields = root.elements();
+        TreeNode root = p.getCodec().readTree(p);
+        if (!(root instanceof ArrayNode)) {
+            throw new CamelRouteDeserializerException(
```

Comment:
I use the custom exception to avoid a case when another `IOException`, which was thrown from elsewhere, was caught.
You are correct. When I throw `IOException` there, Jackson wraps it with `JsonProcessingException` which is caught by the original catch. I am going to change it.